README.md

```diff
@@ -212,7 +213,17 @@ If you install from source, you can modify the code to load custom features as f
 
 1. **Generate features:** If you want to generate features in code, you can write a custom features generator function in `chemprop/features/features_generators.py`. Scroll down to the bottom of that file to see a features generator code template.
 2. **Load features:** If you have features saved as a numpy `.npy` file or as a `.csv` file, you can load the features by using `--features_path /path/to/features`. Note that the features must be in the same order as the SMILES strings in your data file. Also note that `.csv` files must have a header row and the features should be comma-separated with one line per molecule.
- 
+
+#### Atomic Features
+
+Similar to the additional molecular features described above, you can also provide additional atomic features via `--atom_descriptors_path /path/to/features` with valid file formats:
+* `.npz` file, where descriptors are saved as 2D array for each molecule in the exact same order as the SMILES strings in your data file.
+* `.pkl` / `.pckl` / `.pickle` containing a pandas dataframe with smiles as index and numpy array of descriptors as columns.
+* `.sdf` containing all mol blocks with descriptors as entries.
+
+The features can either be used via concatenating the new features to the embedded atomic features after the MPNN via `--atom_descriptors descriptor`, or as additional features used during message passing via `--atom_descriptors feature`.
```

Comment:
Which of these is the default behavior? Is it required that the user choose one?

Also, I haven't used the atom descriptors function yet so I'm not familiar. Am I understanding correctly that every location in the molecule has the same additional features? It's not differentiated by atom or bond, but are whole molecule pieces of information, introduced at the atom level so it can participate in the message passing?

Reply:
The default is `None` (not use atom descriptors) which will create a ValueError if `atom_decriptor_path` is supplied. So, the user needs to choose one. Should I clarify that?

The atom descriptors work as additional features for each atom, so they are different for each atom in the molecule. Eg one could supply quantum-mechanical charges or other properties. Depending on how one chooses `--atom_descriptors`, they will be used either before or after message passing. Before the MPNN means that the atomic feature vectors (currently element, formal charge, hybridisation, ...) will be concatenated with the new features and thus a bit longer. After the MPNN means that the information will be concatenated to the atom feature vectors just before aggregation to the molecular values, so usually averaged or summed up. (Which I think is not that useful, because that could also be achieved via molecular descriptors). 

I have never used atom-descriptors in practice, just helped Yanfei with debugging, so I am not sure under which circumstances they are especially useful.

Reply:
I think a clarification would be helpful for the default behavior. Something along the lines of "users must select in which way atom descriptors are used...".

With respect to the values for different atoms, I think I see that now. It wasn't clear to me on first read but I see that it's implied in the 2D array structure. Would make it clearer if the structure of the arrays was given more explicitly (rows for each atom in the molecule in the order the atom appears in smiles, columns for the descriptor).

Reply:
That seems like a good suggestion to me, will clarify that one has to choose between `descriptor` and `feature`. I will think about how to describe the array structure more clear, and will also need to look up the details (I think the atom order is the one of the rdkit mol object not the smiles string, but have to check).

Reply:
If it’s the rdkit object order (seems likely) then we might benefit from adding a script that prints out the order/connectivity for atoms in different molecules so people know how they need to order inputs.

Reply:
Agree. Maybe print a sdf file? That would have connectivities and orders and we could already write a dummy line with properties that users can overwrite with their data?

Reply:
It might be worth linking the release notes for more details, or copying over the information to the readme/tutorial.

https://github.com/chemprop/chemprop/releases/tag/v1.1.0

Reply:
Oh that is very helpful; I think I will link the release notes, since they are very detailed regarding the format, but maybe too long to cover in the readme.

What do you think about a function to write out order/connectivities? Should I implement a little helper function?

Reply:
I think it would depend on what actual workflows look like. Since the descriptors would likely be generated and formatted automatically, it might be more useful to explain how to programmatically ensure the correct order. A helper function to write out the order could be useful for spot checking though.

Reply:
Okay I think I will then just add a more detailed explanation for now.